src/styles/calendar.scss

```diff
@@ -0,0 +1,46 @@
+@use 'variables';
+
+.calendar {
+  width: 706px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+}
+
+@each $day, $number in variables.$start-day {
+  .calendar--start-day-#{$day} {
+    .calendar__day:first-child {
+      margin-left: 101px * ($number - 1);
+    }
+  }
+}
+
+@each $number in variables.$month-length {
+  .calendar--month-length-#{$number} {
+    @for $i from 1 through $number {
+      .calendar__day:nth-child(#{$i} of .calendar__day) {
```

Comment:
The `:nth-child()` selector is used incorrectly here. The correct syntax should be `.calendar__day:nth-child(#{$i})` without `of .calendar__day`.